woocommerce-gateway-ebanx/services/class-wc-ebanx-query-router.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+if ( ! defined('ABSPATH') ) {
+	exit;
+}
+
+class WC_EBANX_Query_Router {
+	private $key = null;
+	private $routes = array();
+
+	/**
+	 * Construct the router using a query string key
+	 *
+	 * @param string $key
+	 */
+	public function __construct($key) {
+		self::validate_key($key);
+		$this->key = $key;
+	}
+
+	/**
+	 * Maps a key value to a handler method
+	 *
+	 * @param  string $key_value The value which triggers this route
+	 * @param  array  $handler   A callable array with instance/class and method
+	 * @return void
+	 */
+	public function map($key_value, $handler) {
+		self::validate_key_value($key_value);
+		self::validate_route($handler);
+
+		$this->routes[$key_value] = $handler;
+	}
+
+	/**
+	 * The dispatcher that will make coffee for you if you ask nicely
+	 * Just call this at the end of your setup
+	 *
+	 * @return void
+	 */
+	public function serve() {
+		$route = $this->find_route_for_key();
+		if ( $route === null ) {
+			// Not found, carry on
+			return;
+		}
+
+		$params = self::read_request_for_route($route);
+
+		// Route it and die
+		call_user_func_array($route, $params);
+		die();
```

Comment:
Use `wp_die()` instead of `die()`.

https://codex.wordpress.org/Function_Reference/wp_die